packages/api/tests/endpoints/test-granules.js

```diff
@@ -890,23 +991,133 @@ test.serial('move a granule with no .cmr.xml file', async (t) => {
-        {}
+      // Validate S3 Objects are where they should be
+      const bucketObjects = await s3().listObjects({
+        Bucket: bucket,
+        Prefix: destinationFilepath,
+      }).promise();
+      t.is(bucketObjects.Contents.length, 1);
+      t.is(bucketObjects.Contents[0].Key, `${destinationFilepath}/${granuleFileName}.txt`);
+
+      const failedBucketObjects = await s3().listObjects({
+        Bucket: secondBucket,
+        Prefix: `${process.env.stackName}/original_filepath`,
+      }).promise();
+      t.is(failedBucketObjects.Contents.length, 1);
+      t.is(failedBucketObjects.Contents[0].Key,
+        (`${process.env.stackName}/original_filepath/${granuleFileName}.jpg`));
+
+      const thirdBucketObjects = await s3().listObjects({
+        Bucket: thirdBucket,
+        Prefix: destinationFilepath,
+      }).promise();
+      t.is(thirdBucketObjects.Contents.length, 1);
+      t.is(thirdBucketObjects.Contents[0].Key, `${destinationFilepath}/${granuleFileName}.md`);
+
+      // check the granule in dynamoDb is updated and files are replaced
+      const updatedGranule = await granuleModel.get({ granuleId: newGranule.granuleId });
+      const updatedFiles = updatedGranule.files;
+
+      t.true(updatedFiles[0].key.startsWith(`${destinationFilepath}/${granuleFileName}`));
+      t.like(newGranule.files[0], omit(updatedFiles[0], ['fileName', 'key', 'bucket']));
+      t.is(updatedFiles[0].bucket, destinations.find(
+        (dest) => updatedFiles[0].fileName.match(dest.regex)
+      ).bucket);
+
+      t.true(updatedFiles[1].key.startsWith(`${destinationFilepath}/${granuleFileName}`));
+      t.like(newGranule.files[1], omit(updatedFiles[1], ['fileName', 'key', 'bucket']));
+      t.is(updatedFiles[1].bucket, destinations.find(
+        (dest) => updatedFiles[1].fileName.match(dest.regex)
+      ).bucket);
+
+      t.deepEqual(newGranule.files[2], updatedFiles[2]);
+
+      // Check that the postgres granules are in the correct state
+      const pgFiles = await getPostgresFilesInOrder(
+        t.context.knex,
+        newGranule,
+        filePgModel,
+        postgresGranuleCumulusId
       );
 
+      // The .jpg at index 2 should fail and have the original object values as
+      // it's assigned `fakeBucket`
+      for (let i = 0; i < 2; i += 1) {
+        const destination = destinations.find((dest) => pgFiles[i].file_name.match(dest.regex));
+        t.is(destination.bucket, pgFiles[i].bucket);
+        t.like(pgFiles[i], {
+          ...omit(newGranule.files[i], ['fileName', 'size']),
+          key: `${destinationFilepath}/${newGranule.files[i].fileName}`,
+          bucket: destination.bucket,
+          file_name: newGranule.files[i].fileName,
+        });
+      }
+      t.like(pgFiles[2], {
+        ...omit(newGranule.files[2], ['fileName', 'size']),
+        file_name: newGranule.files[2].fileName,
+      });
+    }
+  );
+});
+
+test('move an unmigrated granule', async (t) => {
```

Comment:
Sorry, it's not jumping out at me when scanning this test: what do we mean by "unmigrated" granule here?

Reply:
🤦🏻  I can't blame you for that. 

Title updated: 0dc416b50 / 929286926

This test is validating that if a granule hasn't been migrated to postgres that we don't try to move files in postgres.